File: Structural/Composite/Python/folder.py
# folder.py

from file_system_component import FileSystemComponent

class Folder(FileSystemComponent):
  
  def __init__(self, name):
    self.name = name
    self.components = []

  def add_component(self, component):
    self.components.append(component)

  def show_details(self):
    print(f"Folder: {self.name}")
    for component in self.components:
      component.show_details()

File: Structural/Composite/JavaScript/folder.js
// folder.js

const FileSystemComponent = require("./fileSystemComponent");

class Folder extends FileSystemComponent {
  constructor(name) {
    super();
    this.name = name;
    this.components = [];
  }

  addComponent(component) {
    this.components.push(component);
  }

  showDetails() {
    console.log(`Folder: ${this.name}`);
    this.components.forEach(component => component.showDetails());
  }
}

module.exports = Folder;
